Array_Rotation/ArrayRotationUsingTempArray.java

```diff
@@ -0,0 +1,51 @@
+import java.util.Arrays;
+ 
+public class ArrayRotationUsingTempArray
+{
+    //function to left rotate the array by d elements 
+    private static void rotate(int[] arr, int n, int d) 
+    {
+        //it brings down d to the minimum number of rotations we need to make in case d > n
+         d = d%n; 
+        
+        //initialize temp array of size d
+        int[] temp = new int[d];
+        
+        //copy the first d elements to temp array
+        for(int i=0;i<d;i++)
+        {
+            temp[i]=arr[i];
```

Comment:
Always add space around operators (+, -, *, /, <, >, <=, >=, &&, ||, etc) and after `for` & `if`.